File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
class DifferenceOfSquares {
  double squareOfSum(double n) {
    double result = n * (n + 1) / 2;
    return result * result;
  }

  double sumOfSquares(double n) {
    return n * (n + 1) * (2 * n + 1) / 6;
  }

  double differenceOfSquares(double n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

// SquareOfSum return the square of the sum for the amount of natural number defined
func SquareOfSum(limit int) (result int) {
	for i := 1; i <= limit; i++ {
		result += i
	}

	return result * result
}

// SumOfSquares return the sum of squares for the amout of natural numbers defined
func SumOfSquares(limit int) (result int) {
	for i := 1; i <= limit; i++ {
		result += i * i
	}

	return result
}

// Difference return the diference of the square of sum and sum of square for the
// amount of natural numbers defined
func Difference(limit int) (result int) {
	return SquareOfSum(limit) - SumOfSquares(limit)
}
